doc/sphinx/addendum/nsatz.rst

```diff
@@ -5,7 +5,7 @@ Nsatz: tactics for proving equalities in integral domains
 
 :Author: Loïc Pottier
 
-.. tacn:: nsatz
+.. tacn:: nsatz {? with radicalmax := @term strategy := @term parameters := @term variables := @term }
```

Comment:
I'm quite surprised not to see `@one_term` here in place of `@term`. In practice, it really looks like Tactic Notation's `constr` will parse `@one_term`:
```coq
Tactic Notation "foo" constr(bar) := idtac.
Goal True.
foo id 0.
(* Syntax error: [tactic:ltac_use_default] expected after [tactic:tactic] (in [vernac:tactic_command]). *)
```

contrary to what is documented in https://coq.inria.fr/refman/user-extensions/syntax-extensions.html#tactic-notations and what would appear from
https://github.com/coq/coq/blob/bb6e78ef2ca4bf0d686654fa3f66dd780f5be0ac/plugins/ltac/extraargs.mlg#L39
which would seem to indicate that `constr` is interpreted as `lconstr`, i.e. `term`.

@herbelin can you comment?

Reply:
The call to `create_generic_quotation` is for creating `constr:(...)` and it is normal that it takes an `lconstr`.

Where do you see in the manual that `constr` is supposed to be interpreted as `lconstr`?

Reply:
In the table at https://coq.inria.fr/refman/user-extensions/syntax-extensions.html#tactic-notations. So should this be fixed? I didn't manage to properly track down where these entries for tactic notations are defined.

Reply:
In the table, it is written that it is interpreted as a term, as in `exact`? Isn't it clear that it is a `constr` (at level 8)?

Reply:
Thanks for confirming then that it is a `constr` and not an `lconstr`. In the manual, this means that it should be documented as a `one_term`.
`exact` is indeed still documented as taking a `term` but this is because we haven't fixed the grammar in the tactics chapter yet.

Reply:
It's very confusing to map `constr` to `Pcoq.Constr.lconstr` in quotations.  The quotation name should be the same as the grammar symbol; we should deprecate `constr( ...)` and use a new name, e.g. `one_term`--but not in this PR.

doc_grammar assumes quotation names used in mlgs are identical to the associated grammar symbols.  "exact a b c" is a syntax error, so `exact` should have `one_term` in the doc.  I was not entirely aware that mlgs were using quotations in `TACTIC EXTEND` and `ARGUMENT EXTEND`.

In the meantime, it will take some time to find all the `constr` quotations in the mlgs and somehow get the correct mapping.  I'll update the `Tactic Notation` table, too.

Are these quotation symbols fully equivalent or not?

```
\let () = create_generic_quotation "uconstr" Pcoq.Constr.lconstr Stdarg.wit_uconstr
let () = create_generic_quotation "constr" Pcoq.Constr.lconstr Stdarg.wit_constr
let () = create_generic_quotation "open_constr" Pcoq.Constr.lconstr Stdarg.wit_open_constr
```

@herbelin This is an example of a drawback of having multiple mechanisms for specifying syntax which we discussed a few weeks ago.

Reply:
OTOH, the Ltac2 `constr : ( term )` construct uses a `term`.

Reply:
> It's very confusing to map constr to Pcoq.Constr.lconstr in quotations.

At some time, I advocated to @ppedrot that `constr:` should take a (not-l) `constr` so that we can write `constr:x`, `constr:(x,y)` or `constr:(S x)` but the adopted convention was that parenthese were part of the quotation syntax. On this latter basis, it seems intuitive that the inside is an `lconstr` as this would be a bit strange to have to write `constr:((S x))`, no?

> The quotation name should be the same as the grammar symbol

I believe it is the usual antagonism between when a symbol denotes syntax and when it denotes semantics. The `lconstr` vs `constr` differences is mainly for parsing/printing but both have the same semantic rôle.

> ```
> let () = create_generic_quotation "uconstr" Pcoq.Constr.lconstr Stdarg.wit_uconstr
> let () = create_generic_quotation "constr" Pcoq.Constr.lconstr Stdarg.wit_constr
> let () = create_generic_quotation "open_constr" Pcoq.Constr.lconstr Stdarg.wit_open_constr
> ```
They all have a different semantic.

>  I was not entirely aware that mlgs were using quotations in `TACTIC EXTEND` and `ARGUMENT EXTEND`.

I don't think they do. Or, maybe, I misunderstand what you mean by quotations (which, for me, are the `foo:(...)` syntaxes).

Reply:
> On this latter basis, it seems intuitive that the inside is an `lconstr` as this would be a bit strange to have to write `constr:((S x))`, no?

Yes.

Reply:
> > On this latter basis, it seems intuitive that the inside is an `lconstr` as this would be a bit strange to have to write `constr:((S x))`, no?

OK, seems reasonable.

> I don't think they do. Or, maybe, I misunderstand what you mean by quotations (which, for me, are the foo:(...) syntaxes).

Ah, I may have been confused.  These symbols are used inconsistently and thus quite confusing.  I would like to clean that up in a future PR.  I'll let it be for now.